packages/mainsail/source/helpers/parse-units.ts

```diff
@@ -0,0 +1,14 @@
+import { ARK_MULTIPLIER, GWEI_MULTIPLIER, WEI_MULTIPLIER } from "../crypto/constants";
+
+export const parseUnits = (value: number | string, unit = "ark"): string => {
+	switch (unit.toLowerCase()) {
+		case "wei":
+			return (Number(value) * WEI_MULTIPLIER).toString();
+		case "gwei":
+			return (Number(value) * GWEI_MULTIPLIER).toString();
+		case "ark":
+			return (Number(value) * ARK_MULTIPLIER).toString();
+		default:
+			throw new Error(`Unsupported unit: ${unit}. Supported units are 'wei', 'gwei', and 'ark'.`);
+	}
```

Comment:
best to use a bignumber implementation as these values grow large and can cause weird situations with regular `Number` usage